### PR TITLE
tests: ringbuffer: test concurrent operation

### PR DESCRIPTION
**Pull request by Zhaoningx**
Add a testcase to test that read and write the
same buffer at the same time to check the
integrity of data reading and writing.

Signed-off-by: Ningx Zhao <ningx.zhao@intel.com>

### CODE REVIEW

**Review by enjiamai — APPROVED** [review]

File: tests/lib/ringbuffer/sys_ringbuffer/Kconfig [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2020 Nordic Semiconductor ASA
```
Comment (LeiW000) [inline tests/lib/ringbuffer/sys_ringbuffer/Kconfig:1]: Please change "2020" with "2021".

File: tests/lib/ringbuffer/sys_ringbuffer/src/concurrent.c [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+		databuffer2[i] = 2;
+	}
+	/* Try to write data into the ringbuffer */
+	data_write(databuffer2);
+	/* Try to get data from the ringbuffer and check */
+	data_read(output);
+}
+
+/**
+ * @brief Test that prevent concurrent write
```
Comment (LeiW000) [inline tests/lib/ringbuffer/sys_ringbuffer/src/concurrent.c:90]: It would better to change "write" with "writing".

**Review by LeiW000 — CHANGES_REQUESTED** [review]

**Comment by LeiW000** [comment]
In the PR, you also move some tests from "tests/lib/ringbuffer/" to "/ringbuffer/ringbuffer_api". I think we need to update the commit log to reflect it.

By the way, why do you need to do it?

**Comment by Zhaoningx** [comment]
> In the PR, you also move some tests from "tests/lib/ringbuffer/" to "/ringbuffer/ringbuffer_api". I think we need to update the commit log to reflect it.
> 
> By the way, why do you need to do it?

Hi @LeiW000 , the purpose of doing this is to separate the integration test from the module test, but I find that it is not easy. In the module test, each test case maybe was not independent and has a certain dependency, so I return to the original state and add a source file.

**Review by LeiW000 — APPROVED** [review]

**Pull request merged**